File: July-2023/Java/LongestSubarrayof1'sAfterDeletingOneElement.java
// Sliding Window o(n)
//1493. Longest Subarray of 1's After Deleting One Element
class Solution {
    public int longestSubarray(int[] nums) {
        int i = 0, j = 0, lastIndex = -1, result=0;
        
        while(j < nums.length){
            
             if(nums[j]==0){
                 i = lastIndex+1;
                 lastIndex = j;
              }   
             
            result = Math.max(result,j-i);
            j++;
            
        }
        return result;
    }
}

File: July-2023/C++/LongestSubarrayof1'sAfterDeletingOneElement.cxx
// Sliding window O(n)
// 1493. Longest Subarray of 1's After Deleting One Element
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int i = 0;

        int j = 0;
        
        int last_zero_idx = -1;
        
        int result = 0;
        
        while(j < nums.size()) {
            
            if(nums[j] == 0) {
                i = last_zero_idx+1;
                last_zero_idx = j;
            }
            
            result = max(result, j-i);
            j++;
        }
        
        return result;
    }
};
